staging/src/kubevirt.io/api/core/v1/schema.go

```diff
@@ -465,6 +465,13 @@ type Devices struct {
 	// Whether to emulate a TPM device.
 	// +optional
 	TPM *TPMDevice `json:"tpm,omitempty"`
+
+	// The number of PCIe ports available for the VM. A value lower than the
+	// required amount will get bumped by libvirt to same as amount of inserted
+	// PCIe cards.
+	// Defaults to 0.
+	// +optional
+	NumberPciPorts uint8 `json:"numberPciPorts,omitempty"`
```

Comment:
Does it make sense to set the maximal amount by default? (in order to reduce user friction)

Reply:
I don't understand what user friction is in this context, but more PCIe ports cost more memory.

Thus, setting it to 0 seems more conservative.

The user will always be able to plug at least one interface in.

... should the user want more, he'd need to set this parameter.

At least, that was my reasoning - i.e. I don't think we should impact the cluster when using VMs that will never require hotplug.

Reply:
IIUC the user will need to take into account the required PCI ports for the desired disks and network, and hopefully few more slots for future hotplug if ever needed.

For example: I create a VM with two disks and two NICs that requires 4 slots.
A week later I will need to hotplug a disk and another interface, I might have no PCI ports left.
Will I be able to set the `numberPciPorts` when the VMI already running? (to enable hotplugging more devices).
Or I will need to shut down the VM and add the new devices to the spec?